public/app/features/dashboard-scene/scene/DashboardReloadBehavior.ts

```diff
@@ -0,0 +1,105 @@
+import { isEqual } from 'lodash';
+
+import {
+  formatRegistry,
+  isCustomVariableValue,
+  MultiValueVariable,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  VariableDependencyConfig,
+} from '@grafana/scenes';
+import { VariableFormatID } from '@grafana/schema';
+import { appEvents } from 'app/core/core';
+import { getClosestScopesFacade, ScopesFacade } from 'app/features/scopes';
+import { ALL_VARIABLE_VALUE } from 'app/features/variables/constants';
+import { ReloadDashboardEvent } from 'app/types/events';
+
+export interface DashboardReloadBehaviorState extends SceneObjectState {
+  reloadOnParamsChange?: boolean;
+  uid?: string;
+}
+
+export class DashboardReloadBehavior extends SceneObjectBase {
+  private _scopesFacade: ScopesFacade | null = null;
+
+  constructor({ reloadOnParamsChange, uid }: DashboardReloadBehaviorState) {
+    const shouldReload = reloadOnParamsChange && uid;
+
+    super({});
+
+    this.reloadDashboard = this.reloadDashboard.bind(this);
+
+    if (shouldReload) {
+      this.addActivationHandler(() => {
+        this._scopesFacade = getClosestScopesFacade(this);
+
+        this._variableDependency = new VariableDependencyConfig(this, {
+          onAnyVariableChanged: this.reloadDashboard,
+        });
+
+        this._scopesFacade?.setState({
+          handler: this.reloadDashboard,
+        });
+
+        this._subs.add(
+          sceneGraph.getTimeRange(this).subscribeToState((newState, prevState) => {
+            if (!isEqual(newState.value, prevState.value)) {
+              this.reloadDashboard();
+            }
+          })
+        );
+      });
+    }
+  }
+
+  private isEditing() {
+    return this.parent && 'isEditing' in this.parent.state && this.parent.state.isEditing;
+  }
+
+  private isWaitingForVariables() {
+    return sceneGraph.getVariables(this.parent!).state.variables.some((variable) => variable.state.loading) ?? false;
+  }
+
+  private reloadDashboard() {
+    if (!this.isEditing() && !this.isWaitingForVariables()) {
+      const timeRange = sceneGraph.getTimeRange(this);
+
+      const format = formatRegistry.get(VariableFormatID.QueryParam);
+
+      const timeParams = [
+        `from=${timeRange.state.value.from.toISOString()}`,
+        `to=${timeRange.state.value.to.toISOString()}`,
+      ];
+
+      if (timeRange.state.timeZone) {
+        timeParams.push(`timeZone=${timeRange.state.timeZone}`);
+      }
```

Comment:
can remove this if you use timeRange.urlSync.getUrlState

Reply:
Done. TY!